File: modules/token-core-bridge/android/src/main/java/expo/modules/tokencorebridge/TokenCoreBridgeModule.kt
package expo.modules.tokencorebridge

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class TokenCoreBridgeModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("TokenCoreBridge")

    Function("isAvailable") {
      false
    }

    Function("getDefaultFileDir") {
      throw IllegalStateException("Token Core Android bridge is not wired yet.")
    }

    AsyncFunction("callTcxApi") { _: String ->
      throw IllegalStateException("Token Core Android bridge is not wired yet.")
    }
  }
}

File: modules/token-core-bridge/src/TokenCoreBridgeModule.ts
import { NativeModule, requireOptionalNativeModule } from "expo";

declare class TokenCoreBridgeModule extends NativeModule {
  getDefaultFileDir(): string;
  isAvailable(): boolean;
  callTcxApi(hexPayload: string): Promise<string>;
}

export default requireOptionalNativeModule<TokenCoreBridgeModule>("TokenCoreBridge");
